pkg/chunk/openstack/swift_object_client.go

```diff
@@ -0,0 +1,207 @@
+package openstack
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/ncw/swift"
+)
+
+type SwiftObjectClient struct {
+	conn      *swift.Connection
+	cfg       SwitfConfig
+	delimiter rune
+}
+
+// SwitfConfig is config for the Swift Chunk Client.
+type SwitfConfig struct {
+	ContainerName               string        `yaml:"container_name"`
+	Domain                      string        `yaml:"domain"`                // User's domain name
+	DomainId                    string        `yaml:"domain_id"`             // User's domain Id
+	UserName                    string        `yaml:"user_name"`             // UserName for api
+	UserId                      string        `yaml:"user_id"`               // User Id
+	ApiKey                      string        `yaml:"api_key"`               // Key for api access
+	ApplicationCredentialId     string        `yaml:"app_credential_id"`     // Application Credential ID
+	ApplicationCredentialName   string        `yaml:"app_crendential_name"`  // Application Credential Name
+	ApplicationCredentialSecret string        `yaml:"app_credential_secret"` // Application Credential Secret
+	AuthUrl                     string        `yaml:"auth_url"`              // Auth URL
+	Retries                     int           `yaml:"retries"`               // Retries on error (default is 3)
+	ConnectTimeout              time.Duration `yaml:"connect_timeout"`       // Connect channel timeout (default 10s)
+	Timeout                     time.Duration `yaml:"timeout"`               // Data channel timeout (default 60s)
+	Region                      string        `yaml:"region"`                // Region to use eg "LON", "ORD" - default is use first region (v2,v3 auth only)
+	AuthVersion                 int           `yaml:"auth"`                  // Set to 1, 2 or 3 or leave at 0 for autodetect
```

Comment:
To keep it consistent with the CLI flag:

```suggestion
	AuthVersion                 int           `yaml:"auth_version"`          // Set to 1, 2 or 3 or leave at 0 for autodetect
```